tests/spec/int_series/test_add.py

```diff
@@ -0,0 +1,32 @@
+from databuilder.query_language import Dataset, IdSeries, IntSeries, build_patient_table
+
+from ..helpers import transpose
+
+p = build_patient_table(
+    "p",
+    {
+        "patient_id": IdSeries,
+        "i": IntSeries,
+    },
+)
+
+
+def test_add(in_memory_engine):
+    in_memory_engine.setup(
+        {
+            p: (
+                [1, 101],
+                [2, None],
+            ),
+        }
+    )
+
+    dataset = Dataset()
+    dataset.set_population(True)
+    dataset.v1 = p.i + 1
+    dataset.v2 = 1 + p.i
+
+    results = transpose(in_memory_engine.extract(dataset))
+
+    assert results["v1"] == {1: (101 + 1), 2: None}
+    assert results["v2"] == {1: (1 + 101), 2: None}
```

Comment:
> Am I right to think that `.extract` records the operations that were performed, rather than actually performs the operations?

No, `.extract` actually extracts the data.

> I'm unsure how to read `{1: (101 + 1) ... }`.

`assert results["v1"] == {1: (101 + 1), 2: None}` means that in column v1, we expect patient 1 to have the value 102 and patient 2 to have the value None.  I'd written "101 + 1" instead of "102" to indicate where the value comes from -- perhaps that's unnecessary here, but more helpful in [some of the more involved computations](https://github.com/opensafely-core/databuilder/pull/401/files#diff-fe7e861d2a9478408ebcf6cf7f8ee1ab0443d6e2aac97b7f28fb6e8a2eb5dd4dR33-R36).

Reply:
Thanks for explaining: I'd not appreciated that `(101 + 1)` was performing this role. As elsewhere, once one knows the conventions, it becomes easier to read the tests.